### PR TITLE
chore: use ads icons

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description
Why did this PR appear? I wanted to replace the icons from `ads-old` with the icons from `ads`. After that, I had to fix some affected components in `ads-old` and in the main app. In the process, I discovered that a large amount of code is simply not being used. 

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10419944222>
> Commit: 86491f43aff37e34468fb7dc32722b9ef2ec60c9
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10419944222&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Fri, 16 Aug 2024 13:28:03 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Improved precision in UI element targeting for enhanced test reliability.
	- Broadened icon options within the FilePickerComponent for greater flexibility.

- **Improvements**
	- Shifted component imports to a centralized design system for better consistency.
	- Simplified prop structures for the Button component to enhance clarity.
	- Enhanced validation utilities available for form handling and input validation.

- **Bug Fixes**
	- Enhanced robustness of element locators, reducing potential UI interaction issues.

- **Refactor**
	- Adjusted component properties to align with updated design guidelines, promoting semantic usage.
	- Consolidated exports to emphasize type definitions and utility functions for better maintainability.

- **Chores**
	- Cleaned up imports and updated code structure for improved maintainability.

### CODE REVIEW

File: app/client/packages/design-system/ads-old/src/DisplayImageUpload/index.tsx
Comment (KelvinOm) [inline app/client/packages/design-system/ads-old/src/DisplayImageUpload/index.tsx:1]: Seems we need to move `DisplayImageUpload` to ads package and reuse here `Modal` component from `ads`. After that we'll able delete `DialogComponent` that used only in `DisplayImageUpload`.

cc @albinAppsmith

File: app/client/packages/design-system/ads-old/src/Dropdown/index.tsx [hunk 1]
```diff
@@ -1,27 +1,5 @@
...
-import "./styles.css";
-import { importSvg } from "../utils/icon-loadables";
-
-const Check = importSvg(
-  async () => import("../assets/icons/control/checkmark.svg"),
-);
+import type { IconSizes, IconNames } from "@appsmith/ads";
+import type { SubTextPosition } from "../types/common";
 
 export type DropdownOnSelect = (
```
Comment (KelvinOm) [inline app/client/packages/design-system/ads-old/src/Dropdown/index.tsx:4]: This really odd, but we only use types from here. We need to use the types from ADS, and delete all this. 

cc @albinAppsmith

File: app/client/packages/design-system/ads-old/src/DisplayImageUpload/index.tsx
Comment (albinAppsmith) [reply app/client/packages/design-system/ads-old/src/DisplayImageUpload/index.tsx:1]: Sounds good

**Review by hetunandu — APPROVED** [review]

File: app/client/packages/design-system/ads-old/src/utils/emailValidator.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import {
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/ads-old/src/utils/emailValidator.ts:1]: Right. But there is no such package yet. I will move it in one of the following PRs.

File: app/client/packages/design-system/ads-old/src/index.ts [hunk 3]
```diff
@@ -26,8 +20,12 @@ export * from "./DisplayImageUpload";
 export { default as DraggableList } from "./DraggableList";
 export * from "./DraggableList";
 
-export { default as Dropdown } from "./Dropdown";
-export * from "./Dropdown";
+// export { default as Dropdown } from "./Dropdown";
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/ads-old/src/index.ts:23]: Fixed

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**